content/articles/building-a-custom-stopwatch-npm-library-in-typescript/index.md

```diff
@@ -0,0 +1,313 @@
+
+To install Typescript globally, open the command terminal and run the command below:
+
+```bash
+npm install -g typescript
+```
+
+That should get do the trick.
+
+### Step 2 – Initializing and setting up the folder structure
+A good Typescript package begins with a proper folder structure. To set up your library folder, you need to search and ensure that the package name of your Typescript library does not already exist on the NPM registry. Click on the [NPM registry](https://www.npmjs.com/) to search and select a unique package name.
+
+Once you’ve found a unique package name, we are good to go. Run the command below to create a folder and initialize a `package.json`:
+
+```bash
+mkdir my-custom-stopwatch
+cd my-custom-stopwatch
+
+npm init -y
+```
+
+Replace `my-custom-stopwatch` with your unique package name. That should get your folder ready and a new `package.json` initialized.
+
+### Step 3 – Setting up the package.json file
+The `package.json` is more like the brain of our project, it contains and provides an array of commands, dependencies, etc. in our project. We will set up the `package.json` file by pushing the code snippet below:
+
+```JSON
+{
+  "name": "my-custom-stopwatch",
+  "version": "1.0.0",
+  "description": "an NPM typescript package",
+  "main": "dist/index.js",
+  "files": ["/dist"],
+  "types": "dist/index.d.ts",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "devDependencies": {
+    "react": "^17.0.2",
+    "typescript": "^4.5.4"
+  },
+  "keywords": [],
+  "author": "Typescript developer",
+  "license": "ISC"
+}
+```
+
+### Step 4 – Installing and setting up the required dependencies
+To build and publish our Typescript library, we need to install the Typescript package into our project so the development process can run smoothly. Open the command terminal and run the command below to install Typescript:
+
+```bash
+npm install require=dev typescript
+```
+
+We are installing Typescript as a dev dependency because we are using it for development-only purposes, meaning the user does not need to install Typescript to consume the package.
+
+Once that is completed, it is time to create and set up the Typescript config file. To do that, run the command below:
+
+```bash
+npx tsc --init
+```
+
+That should create and initialize a `tsconfig.json` file. Now let’s set it up by clearing the default options and pasting the snippet shown below:
+
+```JSON
+{
+  "compilerOptions": {
+    "target": "es6",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "allowJs": true,
+    "declaration": true,
+    "sourceMap": true,
+    "module": "esnext",
+    "outDir": "./dist",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "noEmit": true,
+    "jsx": "react"

```

Comment:
`"noEmit": true,` property makes the compile command to produce no output. Please fix this. Remove the jsx option.